File: app/anynode/app/src/nexusMem.ts
import fetch from "node-fetch";

export interface RememberReq {
  role?: string;
  text: string;
  tags?: string[];
  priority?: number;
  ttl_seconds?: number;
}

export class NexusMemClient {
  base: string;
  timeoutMs: number;

  constructor(baseUrl: string = "[REDACTED-URL] timeoutMs: number = 2000) {
    this.base = baseUrl.replace(/\/$/, "");
    this.timeoutMs = timeoutMs;
  }

[REDACTED-SECRET-LINE]
    const u = new URL(this.base + "/context");
[REDACTED-SECRET-LINE]
    if (topics && topics.length) u.searchParams.set("topics", topics.join(","));
    const r = await fetch(u.toString(), {method:"GET"});
    if (!r.ok) throw new Error(`context: ${r.status}`);
    return r.json();
  }

  async remember(req: RememberReq): Promise<any> {
    const r = await fetch(this.base + "/remember", {
      method:"POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({role:"system", priority:5, ...req})
    });
    if (!r.ok) throw new Error(`remember: ${r.status}`);
    return r.json();
  }

  async mirror(text: string, tags: string[] = [], priority = 8, vector?: number[]): Promise<any> {
    const r = await fetch(this.base + "/mirror", {
      method:"POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ text, tags, priority, vector })
    });
    if (!r.ok) throw new Error(`mirror: ${r.status}`);
    return r.json();
  }

  async health(): Promise<any> {
    const r = await fetch(this.base + "/healthz");
    if (!r.ok) throw new Error(`health: ${r.status}`);
    return r.json();
  }
}

File: app/anynode/app/src/nexus_mem.py
import os
from typing import List, Dict, Any, Optional

try:
    import httpx as _http
except Exception:  # pragma: no cover
    import requests as _http  # type: ignore

DEFAULT_BASE = os.getenv("MEM_BASE_URL", "http://127.0.0.1:8303")

class NexusMemClient:
    def __init__(self, base_url: str = DEFAULT_BASE, timeout: float = 2.0):
        self.base_url = base_url.rstrip("/")
        self.timeout = timeout

    def get_context(self, budget_tokens: int = 800, topics: Optional[List[str]] = None) -> Dict[str, Any]:
        params = {"budget_tokens": str(budget_tokens)}
        if topics:
            params["topics"] = ",".join(topics)
        url = f"{self.base_url}/context"
        resp = _http.get(url, params=params, timeout=self.timeout)
        resp.raise_for_status()
        return resp.json()

    def remember(self, text: str, tags: Optional[List[str]] = None, priority: int = 5, ttl_seconds: Optional[int] = None) -> Dict[str, Any]:
        payload = {"role": "system", "text": text, "priority": priority}
        if tags: payload["tags"] = tags
        if ttl_seconds is not None: payload["ttl_seconds"] = ttl_seconds
        url = f"{self.base_url}/remember"
        resp = _http.post(url, json=payload, timeout=self.timeout)
        resp.raise_for_status()
        return resp.json()

    def mirror(self, text: str, vector: Optional[List[float]] = None, tags: Optional[List[str]] = None, priority: int = 8) -> Dict[str, Any]:
        payload = {"text": text, "tags": tags or [], "priority": priority}
        if vector: payload["vector"] = vector
        url = f"{self.base_url}/mirror"
        resp = _http.post(url, json=payload, timeout=self.timeout)
        resp.raise_for_status()
        return resp.json()

    def health(self) -> Dict[str, Any]:
        resp = _http.get(f"{self.base_url}/healthz", timeout=self.timeout)
        resp.raise_for_status()
        return resp.json()
